packages/@o3r/extractors/src/core/comparator/metadata-comparator.interface.ts

```diff
@@ -0,0 +1,76 @@
+import type { JsonObject } from '@angular-devkit/core';
+import type { SupportedPackageManagers } from '@o3r/schematics';
+
+/**
+ * Interface of the comparator used to compare 2 different versions of the same metadata file.
+ */
+export interface MetadataComparator<T> {
+  /**
+   * Get an array of metadata items to parse a metadata file content.
+   * @param content Content of a metadata file
+   */
+  getArray(content: any): T[];
+
+  /**
+   * Get a description of a metadata item.
+   * @param item Metadata item
+   */
+  getName(item: T): string;
+
+  /**
+   * Compares 2 metadata items.
+   * @param item1 Metadata item
+   * @param item2 Metadata item to compare with
+   */
+  isSame(item1: T, item2: T): boolean;
+
+  /**
+   * Returns true if a migration item matches a metadata item.
+   * @param metadataItem Metadata item
+   * @param migrationItem Migration item
+   */
+  isMigrationDataMatch(metadataItem: T, migrationItem: any): boolean;
```

Comment:
please avoid `any`

Reply:
What should I use if I don't know the type of metadata that I will get here ? I could add a generic type like MetadataComparator<T, K> but it seems a bit too much

Reply:
I agree with Kilian we should go with `MetadataComparator<T, K>`